app/routes/problem_routes.py

```diff
@@ -27,3 +28,33 @@ def get_problem_by_id(problem_id):
     
     return jsonify(problem), 200
 
+# Used by Leetcode bot to set a problem as solved
+@bp.route('/<int:problem_id>/solved', methods=['PUT'])
+def mark_problem_solved(problem_id):
+    if not problem_id:
+        return jsonify({'error': 'Problem ID not provided'}), 400
+
+    success = firestore_wrapper.update_last_asked_timestamp(problem_id, datetime.now())
+
+    if not success:
+        return jsonify({'error': f'Cannot mark {problem_id} as solved'}), 400
+    
+    return jsonify({'message': f'Marked {problem_id} as solved'}), 200
+
```

Comment:
Two issues standout to me about this endpoint

1. We need to mark the problem solved for a specific user. So this endpoint should pass the user's discord id in the request body

2. I think this function `firestore_wrapper.update_last_asked_timestamp` only updates the question field of when it was last ask. And this is solely for challenges purpsoes. Instead, you need to update the question and users documents to add them to each other/add another function in firestore if not supported